cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -248,6 +248,57 @@ var moveTests = []struct {
 		},
 		wantErr: false,
 	},
+	{
+		name: "Cluster with MachineDeployment with a static bootstrap secret",
+		fields: moveTestsFields{
+			objs: test.NewFakeCluster("ns1", "cluster1").
+				WithMachineDeploymentsWithoutBootstrapConfigRef(
+					test.NewFakeMachineDeployment("md1").
+						WithMachineSets(
+							test.NewFakeMachineSet("ms1").
+								WithMachines(
+									test.NewFakeMachine("m1"),
+									test.NewFakeMachine("m2"),
+								),
```

Comment:
This seems a little bit of topic for this PR, but I'm not sure if I got right what we intend with "static bootstrap secret", because If I'm not wrong this is going to create a bootstrap config object (which I can see below at L289-290), which usually does not exist when a bootstrap secret is provided.
The same applies to the test case for the object graph